pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -255,6 +255,14 @@ public boolean isMutableSegment() {
 
     // Initialize for each column
     for (FieldSpec fieldSpec : _physicalFieldSpecs) {
+      String fieldSpecName = fieldSpec.getName();
+      if (metricsAggregators.containsKey(fieldSpecName)) {
+        int maxLength = metricsAggregators.get(fieldSpecName).getRight().getMaxAggregatedValueByteSize();
+        if (maxLength > 0) {
+          fieldSpec.setMaxLength(maxLength);
```

Comment:
This will update the original schema, and can potentially cause unexpected behavior. Ideally schema should not be modified